File: socketrunner_unity/Assets/Scripts/Commands/Tree2command.cs
using System;
using UnityEngine;

public class Tree2Command : ICommand
{
	private readonly bool _flag;
	private Tree2 _tree2;

	public Tree2Command(bool flag)
	{
		_flag = flag;
	}

	public void Execute()
	{
		_tree2?.setActiveTree2(_flag);
	}
}

File: server/src/commands/Tree2Command.js
const { Command } = require('./Command');

class Tree2Command extends Command {
    constructor(flag) {
        super();
        this._flag = true;
    }

    get type() {
        return 'TREE2_COMMAND';
    }

    switchFlag(){
        this._flag = !this._flag;
        return this._flag;
    }

    execute(ws) {

        if (!ws || ws.readyState !== 1) return false;
        console.log('reached execute tree2 js');
        ws.send(JSON.stringify({ type: this.type, payload: { flag: this.switchFlag() } }));


        return true;
    }
}

module.exports = { Tree2Command };
